### DISCUSSION TITLE
Give this component a copy of its own tok2vec

### DISCUSSION THREAD

**Discussion by koayst** [explosion/spaCy#9554] [Help: Coding & Implementations] [answered]
Hello,

This question is related to: projects/tutorials/ner_double/.  Part of the code as shown below.

import spacy

nlp = spacy.load("en_core_web_md") # load the base pipeline
drug_nlp = spacy.load("en_ner_drugs") # load the drug pipeline

give this component a copy of its own tok2vec
drug_nlp.replace_listeners("tok2vec", "ner", ["model.tok2vec"])

The last comment said "give this component a copy of its own tok2vec".  Isn't a tok2vec object of ["model.tok2vec"]  is instantiated to replace the tok2vec layer in "ner" component ?

I did a drug_nlp.pipe_names and it returned -> ['tok2vec', 'ner'].

**Comment by koayst** [comment, marked as answer]
I think I know the answer.

Looking at the config.cfg file again.  My understanding now is "ner_drug" uses another tok2vec component.  There are 2 independent tok2vec components loaded into the pipeline.  One tok2vec for "nlp" and other tok2vec for drug_nlp.

Please comment if my understanding is correct.

**Reply by polm** [reply]
That's correct!

Statistical components depend on the tok2vec they were trained with, so if you move them to another pipeline you need to bring the tok2vec along.
